src/main/java/baritone/process/CraftingProcess.java

```diff
@@ -0,0 +1,473 @@
+        if (canCraft(item, amount)) {
+            this.amount = amount;
+            logDirect("Crafting now " + amount + "x [" + recipe.getRecipeOutput().getDisplayName() + "]");
+            if (!canCraftInInventory(recipe)) {
+                getACraftingTable();
+            }
+        } else {
+            logDirect("Insufficient resources.");
+        }
+    }
+
+    @Override
+    public void craftRecipe(IRecipe recipe, int amount) {
+        if (canCraft(recipe, amount)) {
+            this.recipe = recipe;
+            this.amount = amount;
+            logDirect("Crafting now " + amount + "x [" + recipe.getRecipeOutput().getDisplayName() + "]");
+            if (!canCraftInInventory(recipe)) {
+                getACraftingTable();
+            }
+        } else {
+            logDirect("Insufficient resources.");
+        }
+    }
+
+    @Override
+    //should this be in a helper class?
+    public boolean hasCraftingRecipe(Item item) {
+        ArrayList<IRecipe> recipes = getCraftingRecipes(item);
+        return !recipes.isEmpty();
+    }
+
+    @Override
+    //should this be in a helper class?
+    public ArrayList<IRecipe> getCraftingRecipes(Item item) {
+        ArrayList<IRecipe> recipes = new ArrayList<>();
+        for (IRecipe recipe : CraftingManager.REGISTRY) {
+            if (recipe.getRecipeOutput().getItem().equals(item)) {
+                recipes.add(recipe);
+            }
+        }
+        return recipes;
+    }
+
+    @Override
+    //should this be in a helper class?
+    public boolean canCraft(Item item, int amount) {
+        List<IRecipe> recipeList = getCraftingRecipes(item);
+        for (IRecipe recipe : recipeList) {
+            if (canCraft(recipe, amount)) {
+                this.recipe = recipe;
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    //should this be in a helper class?
+    public boolean canCraft(IRecipe recipe, int amount) {
+        RecipeItemHelper recipeItemHelper = new RecipeItemHelper();
+        for (ItemStack stack : ctx.player().inventory.mainInventory) {
+            recipeItemHelper.accountStack(stack);
+        }
+        int outputCount = recipe.getRecipeOutput().getCount();
+        int times = amount % outputCount == 0 ? amount / outputCount : (amount / outputCount) + 1;
+        return recipeItemHelper.canCraft(recipe, null, times);
+    }
+
+    @Override
+    public boolean canCraftInInventory(IRecipe recipe) {
+        return recipe.canFit(2,2) && !ctx.player().isCreative();
+    }
+
+    private void moveItemsToCraftingGrid() {
+        clearToPush = false;
+        int windowId = ctx.player().openContainer.windowId;
+        //try to put the recipe the required amount of times in to the crafting grid.
+        for (int i = 0; i * recipe.getRecipeOutput().getCount() < amount; i++) {
+            mc.playerController.func_194338_a(windowId, recipe, GuiScreen.isShiftKeyDown(), ctx.player());
```

Comment:
Since Baritone's `IPlayerController` class doesn't have this function, you should probably just send the packet directly with `ctx.player().connection.sendPacket(...)`.